### PR TITLE
sqlitebrowser: use qmake for darwin

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.5 19F96
Xcode 11.5 11E608c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by RJVB** [comment]
How is this an enhancement, given that CMake is in every aspect a better build system than qmake? I'm not particularly pleased by the name change of the app bundle; even if this is the "official" name we must assume that users have become accustomed to `SQLiteBrowser.app`.

Apparently the qmake system takes care of bundling the app, removing the need for using the App PortGroup and that would be a reason to use qmake, given the workaround I had to implement to avoid capitalisation. That could be a reason to use qmake if the generated binary is identical (imposing a MacPorts build with the intended compiler and compiler options is more complicated with qmake).
Did you check if the generated app bundle is NOT a standalone one, i.e. that all required Qt frameworks and plugins aren't copied into the bundle? If that's that case you'll need to remove them, possibly replacing them with symlinks if required.

If it's the use of the App PG that bothers you: CMake is capable to generate app bundles too, the same "skeleton" type generated also by the App PG (all KDE apps are generated that way). It should be relatively trivial to patch the appropriate CMakeLists.txt file to achieve that.

**Comment by i0ntempest** [comment]
Indeed the capitalization workaround and the app pg is what bothers me. Also no, the app built is only 6.4MB with no frameworks inside. I'll explore how to use cmake to make app bundle and update this if I can (I'm still new to these stuff). Thanks for the info.

**Comment by kencu** [comment]
One thing to know is that Marcus put a great deal of effort into the qmake PortGroup -- it essentially builds a customized, macports-optimized cache file for every qmake or qmake5 build, and overrides much of the default configure logic with macports nuances. 

So in that way, the builds that come out of the qmake* portgroups could well be superior.

**Comment by RJVB** [comment]
>So in that way, the builds that come out of the qmake* portgroups could well be superior.

I wouldn't know how to quantify that, but I tend to doubt it. I'm well aware of the hoops you have to jump through with qmake, and with the fact that it goes against Qt guidelines to do so.

Ultimately I can't say I care here, though... and if the OP wants to take over maintainership of this port, s/he is welcome to do so. I hardly ever use the app and try to spend more time away from the computer these days.

**Comment by kencu** [comment]
well, @i0ntempest -- you've been offered total control over this port, and @RJVB is happy to step away and leave it to you. Are you taking over?

**Comment by i0ntempest** [comment]
Sorry, was kinda busying doing university stuff the last week. And sure I'll have this port, I'll remove him and add myself as maintainer in the next push, after maybe a little more tweaking to the build process.

**Comment by i0ntempest** [comment]
Okay this is now ready for review, sorry for taking so long.

**Comment by kencu** [comment]
I tried your PR out and it all worked as expected. I don't have a system that can test the linux side out, and not sure if @RJVB is available to do that.

File: databases/sqlitebrowser/Portfile [hunk 1]
```diff
@@ -45,59 +42,64 @@ github.livecheck.regex  {([0-9.]+)}
...
+    qt5.depends_component \
+                    qtmacextras qttools
     cmake.install_rpath-prepend \
                     ${qt_libs_dir} ${prefix}/lib
     configure.args-append \
-                    -DCMAKE_PREFIX_PATH=${prefix}
-    configure.args-replace \
-                    -DQT_LIBRARY_DIR=${qt_frameworks_dir} \
+                    -DCMAKE_PREFIX_PATH=${prefix} \
+                    -DQSCINTILLA_INCLUDE_DIR=${qt_includes_dir} \
```
Comment (RJVB) [inline databases/sqlitebrowser/Portfile:98]: Maybe align this properly; somehow I missed that in my original version?

Comment (i0ntempest) [reply databases/sqlitebrowser/Portfile:98]: Uh I don't see what's misaligned here? Or you mean they should be indented 25 spaces like other lines?

**Comment by RJVB** [comment]
>Uh I don't see what's misaligned here? Or you mean they should be indented 25 spaces like other lines?

OK, never mind than, the diff view suggested it was misaligned but that must have been an effect of wrapping... My bad, sorry.

**Pull request merged**